packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,300 @@
+/// <reference lib="dom" />
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import { TelegramCloudStorageStamperError } from "./errors";
+
+declare global {
+  interface Window {
+    Telegram: any;
+  }
+}
+
+// config when telegram stamper is passed in an API key
+export type TTelegramCloudStorageStamperConfig = {
+  cloudStorageAPIKey?: CloudStorageAPIKey;
+  cloudStorageKey?: string;
+};
+
+export type CloudStorageAPIKey = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+};
+
+// Constant for default key name
+const DEFAULT_TURNKEY_CLOUD_STORAGE_KEY = "@turnkey/TURNKEY_API_KEY";
+
+/**
+ * Stamper to use within a `TurnkeyClient`
+ */
+export default class TelegramCloudStorageStamper {
+  // This stamper uses a typical @turnkey/api-key-stamper under the hood and abstracts away the storage of the actual API keys
+  stamper?: ApiKeyStamper | undefined;
+
+  // the constructor checks if we're in a telegram mini app context, and then gives the developer an interface into looking into CloudStorage even without having an API key attached
+  constructor() {
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+  }
+
+  // create a telegram stamper by getting/setting the private/public API key values from/to telegram cloud storage
+  static async create(config?: TTelegramCloudStorageStamperConfig) {
+    let telegramStamper = new TelegramCloudStorageStamper();
+
+    if (!(await telegramStamper.setSigningKey(config))) {
+      throw new TelegramCloudStorageStamperError(
+        "Failed initializing Telegram Cloud Storage Stamper"
+      );
+    }
+
+    return telegramStamper;
+  }
+
+  async stamp(payload: string) {
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check to see that the stamper was initialized
+    if (!this.stamper) {
+      throw new TelegramCloudStorageStamperError(
+        "Cannot stamp with uninitialized telegram stamper, try running .create() or .setSigningKey()"
+      );
+    }
+
+    return await this.stamper.stamp(payload);
+  }
+
+  // set the API key that is used by the telegram cloud storage for signing requests
+  async setSigningKey(config?: TTelegramCloudStorageStamperConfig) {
+    if (!config) {
+      // try to get API key at default location and set the stamper to use that
+      // passing the DEFAULT_TURNKEY_CLOUD_STORAGE_KEY is not necessary since getAPIKey will use that by default
+      let { apiPublicKey, apiPrivateKey } = await this.getAPIKey(
+        DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
+      );
+
+      if (apiPublicKey && apiPrivateKey) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey,
+          apiPrivateKey,
+        });
+        return true;
```

Comment:
I don't think it's necessary to return a boolean, the function doesn't document a return type. I think `return;` (implicitly: `return undefined`) work better? Otherwise we're back to having devs wondering what this boolean means and what they should do with it.

Reply:
removed